internal/resolver/dns/dns_resolver.go

```diff
@@ -316,8 +325,12 @@ func (d *dnsResolver) lookupHost() ([]resolver.Address, error) {
 }
 
 func (d *dnsResolver) lookup() (*resolver.State, error) {
-	srv, srvErr := d.lookupSRV()
-	addrs, hostErr := d.lookupHost()
+	ctxSRV, cancelSRV := context.WithTimeout(d.ctx, ResolvingTimeout)
+	defer cancelSRV()
```

Comment:
Could you just use (and overwrite) `ctx` for all of these?

```go
ctx, cancel := context.WithTimeout(d.ctx, ResolvingTimeout)
defer cancel()
d.lookupSRV(ctx)
ctx, cancel = context.WithTimeout(d.ctx, ResolvingTimeout)
defer cancel
d.lookupHost(ctx)
// same with TXT
```

(It would probably be ideal to do these calls in parallel so we only need to make the context once, but that would be a bigger behavior change that there's no need to introduce here.)